src/apps/chifra/pkg/file/empty.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2021 The TrueBlocks Authors. All rights reserved.
+// Use of this source code is governed by a license that can
+// be found in the LICENSE file.
+
+package file
+
+import "os"
+
+// Empty is a shortcut function to seek to the beginning
+// of a file and truncate it to 0. It does not call
+// file.Sync though, making all operations in-memory only.
+func Empty(file *os.File) (err error) {
```

Comment:
Can't you just open the file with the right mode? os.O_RDWR|os.O_CREATE|os.O_TRUNC

Reply:
The thing is that this would truncate the file when opening it. And I need to check for the lock first. Locks are additional protocol and `file.Open` doesn't care about them.